crates/gpui/src/platform/windows/direct_write.rs

```diff
@@ -0,0 +1,1286 @@
+                continue;
+            };
+            let Some(font_face) = font_face_ref.CreateFontFace().log_err() else {
+                continue;
+            };
+            let Some(postscript_name) = get_postscript_name(&font_face, &self.components.locale)
+            else {
+                continue;
+            };
+            let is_emoji = font_face.IsColorFont().as_bool();
+            let Some(direct_write_features) = self.generate_font_features(font_features).log_err()
+            else {
+                continue;
+            };
+            let font_info = FontInfo {
+                font_family: family_name.to_owned(),
+                font_face,
+                is_system_font,
+                features: direct_write_features,
+                is_emoji,
+            };
+            let font_id = FontId(self.fonts.len());
+            self.fonts.push(font_info);
+            self.font_id_by_postscript_name
+                .insert(postscript_name, font_id);
+            return Some(font_id);
+        }
+        None
+    }
+
+    unsafe fn update_system_font_collection(&mut self) {
+        let mut collection = std::mem::zeroed();
+        self.components
+            .factory
+            .GetSystemFontCollection2(false, &mut collection, true)
+            .unwrap();
+        self.system_font_collection = collection.unwrap();
+    }
+
+    fn select_font(&mut self, target_font: &Font) -> FontId {
+        let family_name = if target_font.family == ".SystemUIFont" {
+            // https://learn.microsoft.com/en-us/windows/win32/uxguide/vis-fonts
+            // Segoe UI is the Windows font intended for user interface text strings.
+            "Segoe UI"
+        } else {
+            target_font.family.as_ref()
+        };
+        unsafe {
+            // try to find target font in custom font collection first
+            self.get_font_id_from_font_collection(
+                family_name,
+                target_font.weight,
+                target_font.style,
+                &target_font.features,
+                false,
+            )
+            .or_else(|| {
+                self.get_font_id_from_font_collection(
+                    family_name,
+                    target_font.weight,
+                    target_font.style,
+                    &target_font.features,
+                    true,
+                )
+            })
+            .or_else(|| {
+                self.update_system_font_collection();
+                self.get_font_id_from_font_collection(
+                    family_name,
+                    target_font.weight,
+                    target_font.style,
+                    &target_font.features,
+                    true,
+                )
+            })
+            .or_else(|| {
+                log::error!("{} not found, use Zed Mono instead.", family_name);
+                self.get_font_id_from_font_collection(
+                    "Zed Mono",
```

Comment:
GPUI shouldn't depend on Zed fonts, can we change this to some known default? Is Helvetica safe to use?

Reply:
You're right, I always have `Zed` on my mind, forgot that we are on GPUI actually. I opted for `Arial` as the default font because I found that `Helvetica` is missing in my Windows 11 Home Chinese version.